torch/quantization/convert_modules.py

```diff
@@ -0,0 +1,149 @@
+from __future__ import absolute_import, division, print_function, unicode_literals
+import torch.nn as nn
+import torch.nn.quantized as nnq
+import torch
+from .observer import *
+
+class QuantWrapper(nn.Module):
+    def __init__(self, module, qconfig):
+        super(QuantWrapper, self).__init__()
+        self.quant = QuantStub(qconfig)
+        self.dequant = DeQuantStub()
+        self.module = module
+
+    def forward(self, X):
+        X = self.quant(X)
+        self.module.forward(X)
+        X = self.dequant(X)
+        return X
+
+def _observer_forward_hook(self, input, output):
+    self.observer(output)
+
+def add_observer(module, qconfig_dict, qconfig_parent=None, prefix=''):
+    r"""Transforms a module to a quantized module according to qconfig_dict.
+
+    This function insert observer module to all leaf child module of a given
+    module based on qconfig_dict.
+
+    Args:
+        module: instance of the module we want to transform
+        qconfig_dict: dictionary that maps from name of submodule to quantization
+                     configuration
+        qconfig_parent: quantization config of parent module, we will fallback to
+                       this config when there is no specified config for current
+                       module
+        prefix: corresponding prefix of the current module, used as key in
+                qconfig_dict
+    """
+    assert not hasattr(module, 'qconfig'), 'Original module should not have \
+        qconfig object already'
+    module.qconfig = None
+    if prefix in qconfig_dict:
+        module.qconfig = qconfig_dict[prefix]
+    else:
+        module.qconfig = qconfig_parent
+
+    # Insert observers only for leaf nodes, note that this observer is for
+    # the output of the module, for input QuantStub will observe them
+    if module.qconfig is not None and len(module._modules) == 0:
+        # observer and hook will be gone after we swap the module
+        module.add_module('observer', module.qconfig.activation())
+        module.register_forward_hook(_observer_forward_hook)
+        return QuantWrapper(module, module.qconfig)
```

Comment:
wait, didn't we agree to separate observer insertion and quant_dequant? what if I just want to insert QuantStub on my own? One way to structure add_quant_dequant API is to operate only on modules supported in REMAPPING and whose names are in qconfig subtree

Reply:
Oh right, I'll split

Reply:
So, is the plan to have two functions:
 - Add observer that only inserts observers
 - Insert Q DQ that would insert Q and DQ stubs before/after each module within a specified sub-module in QconfigDict.
If whole module is true, then  inserting Q/DQ should happen only at the beginning and end of each sub-module.